src/radical/pilot/agent/launch_method/srun.py

```diff
@@ -129,15 +129,24 @@ def get_launch_cmds(self, task, exec_path):
             # the scheduler did place tasks - we can't honor the core and gpu
             # mapping (see above), but we at least honor the nodelist.
             nodelist = [rank['node_name'] for rank in slots['ranks']]
+            n_nodes  = len(set(nodelist))
 
             # older slurm versions don't accept nodefiles
+            # 42 node is the upper limit to switch from `--nodelist`
+            # to `--nodefile`
             if self._vmajor > 18:
-                nodefile = '%s/%s.nodes' % (sbox, uid)
-                with ru.ru_open(nodefile, 'w') as fout:
-                    fout.write(','.join(nodelist))
-                    fout.write('\n')
+                if n_nodes > 42:
+                    nodefile = '%s/%s.nodes' % (sbox, uid)
+                    with ru.ru_open(nodefile, 'w') as fout:
+                        fout.write(','.join(nodelist))
+                        fout.write('\n')
+                    nodelist.clear()
+
+            # corener case that we can not address, related ticket:
+            # https://github.com/radical-cybertools/radical.pilot/issues/2488
+            elif self._vmajor <= 18 and n_nodes > 42:
```

Comment:
The check for `self._vmajor <= 18` seems redundant - in this else it's always true.  But also, we don't need to raise - we can the LM try with that long nodelist as command line arg anyway: either it works, hurray, or it fails, then we are not worse off than raising now...